### PR TITLE
feat: use Market instead of Symbol in TradeEvent

### PR DESCRIPTION
**Pull request by MaxMoskalenko**

### CODE REVIEW

**Review by maxpushka — CHANGES_REQUESTED** [review]

File: pkg/quotes/market.go [hunk 1]
```diff
@@ -11,11 +13,35 @@ type Market struct {
...
+// NewMarketFromString returns a new Market from a string
+// "btc/usdt" -> Market{BTC, Usdt}
+// NOTE: string should contain "/" delimiter
+func NewMarketFromString(s string) (Market, bool) {
+	parts := strings.Split(s, "/")
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		return Market{}, false
+	}
+	return NewMarket(parts[0], parts[1]), true
+}
```
[the remark this answers is not in the document]
Comment (MaxMoskalenko) [reply pkg/quotes/market.go:38]: It will be used in dependent packages (e.g. neodax, clearport)

Comment (maxpushka) [reply pkg/quotes/market.go:38]: Yeah, just found it out after I looked into a PR for Neodax 😀

File: pkg/quotes/market.go [hunk 2]
```diff
@@ -11,11 +13,35 @@ type Market struct {
...
+func (m Market) String() string {
+	return strings.ToLower(fmt.Sprintf("%s/%s", m.BaseUnit, m.QuoteUnit))
+}
+
+func NewMarket(base, quote string) Market {
+	return Market{
+		BaseUnit:  base,
+		QuoteUnit: quote,
+	}
+}
```
[the remark this answers is not in the document]
Comment (maxpushka) [reply pkg/quotes/market.go:27]: That's right but it also depends on context. The current package design does not force clients to use the constructor, which could lead to inconsistency in how instances of the `Market` type are created throughout the codebase. To force this, getters can be added alongside making the fields private.

Direct struct initialization also has the advantage of making the code more idiomatic to Go, where the language's design encourages simplicity and direct manipulation of struct fields when possible.

[the remark this answers is not in the document]
Comment (MaxMoskalenko) [reply pkg/quotes/market.go:27]: I change both fields to private scope, to prevent (lower/upper)case problems in the future

**Review by maxpushka — APPROVED** [review]

**Pull request merged**